pkg/cephfs/volumemounter.go

```diff
@@ -47,6 +48,101 @@ var (
+// quota support to previous versions. This function checks if the running
+// kernel is one of the versions that have the feature/fixes backported.
+//
+// `uname -r` (or Uname().Utsname.Release has a format like 1.2.3-rc.vendor
+// This can be slit up in the following components:
+// - version (1)
+// - patchlevel (2)
+// - sublevel (3) - optional, defaults to 0
+// - extraversion (rc) - optional, matching integers only
+// - distribution (.vendor) - optional, match against whole `uname -r` string
+//
+// For matching multiple versions, the kernelSupport type contains a backport
+// bool, which will cause matching
+// version+patchlevel+sublevel+(>=extraversion)+(~distribution)
+//
+// In case the backport bool is fase, a simple check for higher versions than
+// version+patchlevel+sublevel is done.
+func kernelSupportsQuota(release string) error {
+	type kernelSupport struct {
+		version      int
+		patchlevel   int
+		sublevel     int
+		extraversion int    // prefix of the part after the first "-"
+		distribution string // component of full extraversion
+		backport     bool   // backports have a fixed version/patchlevel/sublevel
+	}
+
+	quotaSupport := []kernelSupport{
+		{4, 17, 0, 0, "", false},       // standard 4.17+ versions
+		{3, 10, 0, 1062, ".el7", true}, // RHEL-7.7
+	}
+
+	vers := strings.Split(strings.SplitN(release, "-", 2)[0], ".")
+	version, err := strconv.Atoi(vers[0])
+	if err != nil {
+		return fmt.Errorf("failed to parse version from %s: %v", release, err)
+	}
+	patchlevel, err := strconv.Atoi(vers[1])
+	if err != nil {
+		return fmt.Errorf("failed to parse patchlevel from %s: %v", release, err)
+	}
+	sublevel := 0
+	if len(vers) >= 3 {
+		sublevel, err = strconv.Atoi(vers[2])
+		if err != nil {
+			return fmt.Errorf("failed to parse sublevel from %s: %v", release, err)
+		}
+	}
+	extra := strings.SplitN(release, "-", 2)
+	extraversion := 0
+	if len(extra) == 2 {
+		// ignore errors, 1st component of extraversion does not need to be an int
+		extraversion, err = strconv.Atoi(strings.Split(extra[1], ".")[0])
+		if err != nil {
+			// "go lint" wants err to be checked...
+			extraversion = 0
+		}
+	}
+
+	// compare running kernel against known versions
+	for _, kernel := range quotaSupport {
+		if version > kernel.version && !kernel.backport {
+			return nil
+		} else if version == kernel.version {
+			if patchlevel > kernel.patchlevel && !kernel.backport {
+				return nil
+			} else if patchlevel == kernel.patchlevel {
+				if sublevel >= kernel.sublevel && !kernel.backport {
+					return nil
+				} else if sublevel < kernel.sublevel {
+					continue
+				}
+
+				// no sense it matching further if not a backport
+				if !kernel.backport {
+					continue
+				}
+
+				// match distribution
+				if !strings.Contains(release, kernel.distribution) {
+					continue
+				}
+
+				if extraversion >= kernel.extraversion {
+					return nil
+				}
+			}
+		}
```

Comment:
(nit) Readability of code here is a bit convoluted, it works, but suggesting an alternate as below,

```suggestion
        if (!kernel.backport) {
            // deal with the default case(s) when it is not a backport
            if (version >= kernel.version && patchlevel >= kernel.patchlevel && sublevel >= kernel.sublevel) {
                return nil
            }
        } else {
            // specific backport, match distribution initially
            if !strings.Contains(release, kernel.distribution) {
                continue
            }

            if (version >= kernel.version && patchlevel >= kernel.patchlevel && sublevel >= kernel.sublevel && extraversion >= kernel.extraversion) {
                return nil
            }
        }
```

Reply:
This will not work for 5.0.0 release, I will leave the current code  as it is